libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -253,6 +253,8 @@ - (void) runSync:(SFSyncState*) sync updateBlock:(SFSyncSyncManagerUpdateBlock)u
 
     SyncFailBlock failSync = ^(NSString* failureMessage, NSError* error) {
         __strong typeof(weakSelf) strongSelf = weakSelf;
+        //Set error message to sync state
+        [sync setErrorJSON: (NSString*) [[error.userInfo valueForKey:@"error"] objectAtIndex:0]];
```

Comment:
See my comment here: https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1793#discussion_r221398663

The sync error is not necessarily the error returned by the service.
During sync up, if a record doesn't save on the server, we keep going and only network errors or other unexpected failures will cause the sync to fail.

Reply:
Resolved in commit ba068115bbd4cc32a66d8e5f7fd213ba3632e977 & d16984db460575863a9a68d6d0d9a4602ad33b17.
I tried to keep to error.userInfo instead of converting it to string ([error.userInfo description]) so that it returns as JSON on the JS side.
Just wonder how to do if we want to align the error returned in iOS with Android.
Say it also returns JSON instead of string in Android.

Reply:
Resolved in d1f5324ae905e05629c06d5e552f9727b3021f06